gnucash/gnucash-commands.cpp

```diff
@@ -434,3 +547,137 @@ Gnucash::report_list (void)
     scm_boot_guile (0, nullptr, scm_report_list, NULL);
     return 0;
 }
+
+// scripting code follows:
+
+static void
+run_guile_cli (void *data, [[maybe_unused]] int argc, [[maybe_unused]] char **argv)
+{
+    auto args = static_cast<scripting_args*>(data);
+    if (args->script)
+    {
+        PINFO ("Running script from %s... ", args->script->c_str());
+        scm_c_primitive_load (args->script->c_str());
+    }
+    if (args->interactive)
+    {
+        std::cout << _("Welcome to Gnucash Interactive Guile Session") << std::endl;
+        std::vector<const char*> modules =
+            { "gnucash core-utils", "gnucash engine", "gnucash app-utils", "gnucash report",
+              "system repl repl", "ice-9 readline" };
+        auto show_and_load = [](const auto& mod)
+        {
+            std::cout << bl::format ("(use-modules ({1}))") % mod << std::endl;
+            scm_c_use_module (mod);
+        };
+        std::for_each (modules.begin(), modules.end(), show_and_load);
+        scm_c_eval_string ("(activate-readline)");
+        scm_c_eval_string ("(start-repl)");
+    }
+    cleanup_and_exit_with_save (args);
+}
+
+
+int
+Gnucash::run_scripting (std::vector<std::string> m_script_args,
+                        const bo_str& m_file_to_load,
+                        std::string& m_language,
+                        const bo_str& m_script,
+                        bool m_open_readwrite,
+                        bool m_interactive)
+{
+    std::vector<std::string> errors;
+    static const std::vector<std::string> languages = { "guile", "python" };
+
+    if (m_open_readwrite && !m_file_to_load)
+        errors.push_back (_ ("--readwrite: missing datafile!"));
+
+    if (m_script && (!boost::filesystem::is_regular_file (*m_script)))
+        errors.push_back ((bl::format (_("--script: {1} is not a file")) % *m_script).str());
+
+    if (std::none_of (languages.begin(), languages.end(),
+                      [&m_language](auto& lang){ return m_language == lang; }))
+        errors.push_back (_ ("--language: must be 'python' or 'guile'"));
+#ifndef HAVE_PYTHON_H
+    else if (m_language == "python")
+        errors.push_back (_("--language: python wasn't compiled in this build"));
+#endif
+
+    if (!errors.empty())
+    {
+        std::cerr << _("Errors parsing arguments:") << std::endl;
+        auto to_console = [](const auto& str){ std::cerr << str << std::endl; };
+        std::for_each (errors.begin(), errors.end(), to_console);
+        gnc_shutdown_cli (1);
+    }
+
+    std::vector<const char*> newArgv {"gnucash-cli"};
+    std::transform (m_script_args.begin(), m_script_args.end(), std::back_inserter(newArgv),
+                    [](const std::string& s) { return s.c_str(); });
+    // note the vector<const char*> is valid as long as m_script_args's strings are not damaged!
+
+    gnc_prefs_init ();
+    gnc_ui_util_init();
+    if (m_file_to_load)
+        load_file (*m_file_to_load, m_open_readwrite);
```

Comment:
Assign the rv with `[[maybe_unused]]` to avoid compiler warnings.

Reply:
I'm not sure which var is unused here.

Reply:
The rv of `load_file` is unused. As I explained elsewhere, the latest versions of clang and gcc have taken to emitting warnings about discarded rvs. To avoid that you need to say `auto foo{load_file(*file_to_load, open_readwrite)};`; since foo isn't used the compilers will then complain about foo being defined but not used so you need to say `[[maybe_unused]] auto foo{load_file(*file_to_load, open_readwrite)};`.